lib/doh.c

```diff
@@ -182,6 +186,33 @@ doh_write_cb(const void *contents, size_t size, size_t nmemb, void *userp)
   return realsize;
 }
 
+#ifdef USE_HTTPSRR
+static void local_print_buf(struct Curl_easy *data,
+                            unsigned char *buf, size_t len)
+{
+  size_t j;
+  char string_buffer[2048], *ptr;
+
+  ptr = string_buffer;
+  if(len > 1000) {
+    infof(data, "\t:buffer too long %ld", len);
+    return;
+  }
+  for(j = 0; j < len; j ++) {
+    *ptr++ = BIN2HEXDIGIT(buf[j] / 16);
+    *ptr++ = BIN2HEXDIGIT(buf[j] % 16);
```

Comment:
```suggestion
    *ptr++ = BIN2HEXDIGIT(buf[j] >> 4);
    *ptr++ = BIN2HEXDIGIT(buf[j] & 0x0f);
```

Personally I find that better showing that we are rotating and masking out bits.

Reply:
sure, did that